src/games/supported/Adventure.cpp

```diff
@@ -0,0 +1,122 @@
+/* *****************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ */
+#include "Adventure.hpp"
+
+#include "../RomUtils.hpp"
+
+AdventureSettings::AdventureSettings() {
+
+  reset();
+}
+
+/* create a new instance of the rom */
+RomSettings* AdventureSettings::clone() const {
+
+  RomSettings* rval = new AdventureSettings();
+  *rval = *this;
+  return rval;
+}
+
+/* process the latest information from ALE */
+void AdventureSettings::step(const System& system) {
+
+  int chalice_status = readRam(&system, 0xB9);
+  bool chalice_in_yellow_castle = chalice_status == 0x12;
+
+  if (chalice_in_yellow_castle) {
+    m_reward = 1;
+  }
+
+  int player_status = readRam(&system, 0xE0);
+  bool player_eaten = player_status == 2;
+
+  if (player_eaten) {
+    m_reward = -1;
```

Comment:
About removing the -1: of course I can and will do it. On the other hand, you really want to make it even harder? I'd be very curious to see what your intrinsically motivated agent does with this game.